sympy/logic/boolalg.py

```diff
@@ -2305,30 +2306,33 @@ def _rem_redundancy(l1, terms):
 def _input_to_binlist(inputlist, variables):
     binlist = []
     bits = len(variables)
-    for val in inputlist:
-        if isinstance(val, int):
-            binlist.append(ibin(val, bits))
-        elif isinstance(val, dict):
-            nonspecvars = list(variables)
-            for key in val.keys():
-                nonspecvars.remove(key)
-            for t in product((0, 1), repeat=len(nonspecvars)):
-                d = dict(zip(nonspecvars, t))
-                d.update(val)
-                binlist.append([d[v] for v in variables])
-        elif isinstance(val, (list, tuple)):
-            if len(val) != bits:
-                raise ValueError("Each term must contain {bits} bits as there are"
-                                 "\n{bits} variables (or be an integer)."
-                                 "".format(bits=bits))
-            binlist.append(list(val))
-        else:
-            raise TypeError("A term list can only contain lists,"
-                            " ints or dicts.")
+    try:
+        for val in inputlist:
+            if isinstance(val, dict):
+                nonspecvars = list(variables)
+                for key in val.keys():
+                    nonspecvars.remove(key)
+                for t in product((0, 1), repeat=len(nonspecvars)):
+                    d = dict(zip(nonspecvars, t))
+                    d.update(val)
+                    binlist.append([d[v] for v in variables])
+            elif iterable(val):
+                if len(val) != bits:
+                    raise ValueError("Each term must contain {bits} bits as there are"
+                                    "\n{bits} variables (or be an integer)."
+                                    "".format(bits=bits))
+                nums = [operator.index(v) for v in val]
+                binlist.append(nums)
+            else:
+                nums = ibin(operator.index(val),bits=bits)
+                binlist.append(nums)
+    except TypeError:
+        raise TypeError("A term list can only contain lists,"
+                        " ints or dicts.")
```

Comment:
Large blocks of code should not go in the try block of a try/except suite because it can end up catching an error that it should not catch. Only the exact function call that is expected to raise TypeError should be inside the try block. Also there is not much point catching a TypeError just to raise a TypeError.

Reply:
Sir, thank you for the suggestion about using try/except suite in big blocks of code , I will be fixing it . I caught a TypeError and raised another because `operator.index` gives it's own TypeError which just says that the invalid data type is not allowed . We had this TypeError predefined to tell the users the correct data types which are allowed . Although I think it might not be the best way to do it .

Reply:
I have made the changes in the new commit , please check it out .